### ISSUE TITLE
Settings not overriding JAVA_HOME

### ISSUE THREAD

**Issue by edvardfagerholmsc** [zed-extensions/java#9]
If I load a Java program, I get a "Failed to run Java" error in the status bar. I currently have the following configuration set in Zed:

```
{
  "ui_font_size": 16,
  "buffer_font_size": 16,
  "theme": {
    "mode": "system",
    "light": "Andromeda",
    "dark": "One Dark"
  },
  "telemetry": {
    "diagnostics": false,
    "metrics": false
  },
  "lsp": {
    "jdtls": {
      "settings": {
        "java_home": "/Library/Java/JavaVirtualMachines/temurin-21.jdk/Contents/Home"
      }
    }
  }
}
```

However, if I print out the contents of `os.environ` in `/opt/homebrew/Cellar/jdtls/1.40.0/libexec/bin/jdtls.py`, which looks looks to be the file that the java extension is running, I see:

'JAVA_HOME': '/Library/Java/JavaVirtualMachines/temurin-11.jdk/Contents/Home'

In other words, if my shell sets the `JAVA_HOME` environment variable, then that's what is being used regardless of what is configured in my Zed configuration.

**Comment by valentinegb** [comment]
I'm unable to reproduce this on my machine. What versions of Zed and JDTLS are you using?

**Comment by valentinegb** [comment]
Can you please try setting the `settings.java.home` initialization option as explained in the `README.md` of the `initialization-options` branch? (You do *not* need to be using the version of the extension in that branch to set initialization options.)

**Comment by valentinegb** [comment]
#13 has been merged. Please try using `settings.java.home` and if that doesn't work, please [bring it up with the JDTLS developers](https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues).
